tools/file-generator-tool/Microsoft.Health.Dicom.FileGenerator/Program.cs

```diff
@@ -0,0 +1,218 @@
+            getDefaultValue: () => 1);
+
+        numberOfStudies.AddValidator((OptionResult result) =>
+        {
+            if (result.GetValueForOption(numberOfStudies) < 1 || result.GetValueForOption(numberOfStudies) >= MaxStudies)
+            {
+                result.ErrorMessage = $"The value of --studies must not be less than 1 or more than {MaxStudies}.";
+            }
+        });
+
+        var numberOfSeries = new Option<int>(
+            "--series",
+            description: $"Number of series to create per study. Must be between 1 (default) and {MaxSeries}",
+            getDefaultValue: () => 1);
+
+        numberOfSeries.AddValidator((OptionResult result) =>
+        {
+            if (result.GetValueForOption(numberOfSeries) < 1 || result.GetValueForOption(numberOfSeries) > MaxSeries)
+            {
+                result.ErrorMessage = $"The value of --series must not be less than 1 or more than {MaxSeries}.";
+            }
+        });
+
+        var numberOfInstances = new Option<int>(
+            "--instances",
+            description: $"Number of instances to create per series. Must be between 1 (default) and {MaxInstances}",
+            getDefaultValue: () => 1);
+
+        numberOfInstances.AddValidator((OptionResult result) =>
+        {
+            if (result.GetValueForOption(numberOfInstances) < 1 || result.GetValueForOption(numberOfInstances) > MaxInstances)
+            {
+                result.ErrorMessage = $"The value of --instances must not be less than 1 or more than {MaxInstances}.";
+            }
+        });
+
+        var rootCommand = new RootCommand("Generate one or optionally many DICOM file(s).");
+
+        rootCommand.AddOption(filePath);
+        rootCommand.AddOption(largeFile);
+        rootCommand.AddOption(invalidSS);
+        rootCommand.AddOption(invalidDS);
+        rootCommand.AddOption(numberOfStudies);
+        rootCommand.AddOption(numberOfSeries);
+        rootCommand.AddOption(numberOfInstances);
+
+        rootCommand.SetHandler(StoreImageAsync, filePath, largeFile, invalidSS, invalidDS, numberOfStudies, numberOfSeries, numberOfInstances);
+        rootCommand.Invoke(args);
+    }
+
+    private static void StoreImageAsync(string filePath, bool largeFile, bool invalidSS, bool invalidDS, int numberOfStudies = 1, int numberOfSeries = 1, int numberOfInstances = 1)
+    {
+        if (!string.IsNullOrWhiteSpace(filePath) && !Path.EndsInDirectorySeparator(filePath))
+        {
+            filePath += Path.DirectorySeparatorChar;
+        }
+
+        var studyUids = Enumerable.Range(1, numberOfStudies).Select(_ => DicomUIDGenerator.GenerateDerivedFromUUID()).ToList();
+        var seriesUids = Enumerable.Range(1, numberOfSeries).Select(_ => DicomUIDGenerator.GenerateDerivedFromUUID()).ToList();
+        var instanceUids = Enumerable.Range(1, numberOfInstances).Select(_ => DicomUIDGenerator.GenerateDerivedFromUUID()).ToList();
+
+        studyUids.ForEach(
```

Comment:
nit: not blocking. If you care about performance dont use Enumerable.ForEach. This has the worst performance when compared to for and foreach loops.